ejb/src/main/java/org/wildfly/httpclient/ejb/Serializer.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2024 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.httpclient.ejb;
+
+import static org.wildfly.httpclient.ejb.TransactionInfo.LOCAL_TRANSACTION;
+import static org.wildfly.httpclient.ejb.TransactionInfo.NULL_TRANSACTION;
+import static org.wildfly.httpclient.ejb.TransactionInfo.REMOTE_TRANSACTION;
+import static org.wildfly.httpclient.ejb.TransactionInfo.localTransaction;
+import static org.wildfly.httpclient.ejb.TransactionInfo.nullTransaction;
+import static org.wildfly.httpclient.ejb.TransactionInfo.remoteTransaction;
+
+import org.jboss.ejb.client.EJBModuleIdentifier;
+import org.wildfly.transaction.client.SimpleXid;
+
+import javax.transaction.xa.Xid;
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
```

Comment:
"A helper class for serializing and deserializing various data types appearing as invocation patameters"

Reply:
Fixed.